### ISSUE TITLE
Some cities arae not coming - in the osmdata

### ISSUE THREAD

**Issue by Rahulvks** [BlakeRMills/30DayMapChallenge#1]
Hi, Thank you for the wonderful r scripts

some cities are not coming how we add new cities 

`Dubai <- CityStreets("Dubai , United Arab Emirates", "Dubai", "8")`
```

> Dubai <- CityStreets("Dubai , United Arab Emirates", "Dubai", "8")
Warning messages:
1: In structure(unlist(lapply(y, function(yy) fn(x, yy, s2::s2_options(model = model,  :
  Calling 'structure(NULL, *)' is deprecated, as NULL cannot have attributes.
  Consider 'structure(list(), *)' instead.
2: attribute variables are assumed to be spatially constant throughout all geometries 

```

Not sure how you add the min and max lat and long 

`London <- CityStreets(c(-0.7278, 51.2911, 0.3419, 51.744), "London", adminlevel="6")`

Anything i am missing ?

**Comment by Rahulvks** [comment]
Tried this  but still not able to get the min and max like london
```

bb <- getbb('Dubai, united arab emirates' )
q <- opq(bbox = bb)
identical(q, opq(bbox = 'Dubai, united arab emirates'))
```

**Comment by BlakeRMills** [comment]
Hi, so I think the code in question is from Day 5. The function that I wrote works great when you know 1. The City, 2. The admin level of the boundary, and 3. how OSM labels the boundary. I was able to quickly do all of those cities, because I manually adjusted for these.

I would recommend not using the function, and using the individual pieces in the function first. You can also find the administrative boundary number on openstreetmap.org. It looks like Dubai's is "4".

First, find the city boundary, but remove the filter part (since we don't know how it's labeled.

`border <- opq(bbox="Dubai, United Arab Emirates") %>%
    add_osm_feature(key="admin_level", value="4) %>%
    osmdata_sf() %>%
    .$osm_multipolygons %>%
    select(osm_id, name, geometry)`

After this, there should be some filter options. It looks like Dubai is written in the native language, so filtering for "Dubai" won't work, you have to filter for "دبي". OSM is a little unpredictable with their labeling. For instance, filtering for "Kolkata" works but for Rome, you have to filter for "Roma".

The error you're getting is probably because you have the wrong admin level, or you have the wrong filter. Either way, you end up with an empty boundary, and when you try to cut roads based on the empty boundary, it returns an error. 

In short, if you want to add new cities, I would recommend breaking up the function to figure out these pieces this way to get what you are looking for. 

This code worked for me to get it:
Dubai <- CityStreets("Dubai, United Arab Emirates", "دبي", "4")

Let me know if this answers your question.

Best,
Blake

**Comment by Rahulvks** [comment]
Hi Blake, Thank you so much for the explanation and i have fixed the issues in my script based on your inputs, I have re-written the function but i used same colour (your script). One thank you and appreciate.
